library/Zend/Filter/DateTimeSelect.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Filter;
+
+/**
+ * Class DateTimeSelect
+ * @package Zend\Filter
+ */
+class DateTimeSelect extends AbstractDateDropdown
+{
+    /**
+     * Year-Month-Day Hour:Min:Sec
+     *
+     * @var string
+     */
+    protected $format = '%6$s-%4$s-%1$s %2$s:%3$s:%5$s';
+
+    /**
+     * @param mixed $value
+     * @return array|mixed|null|string
+     */
+    public function filter($value)
+    {
+        if (is_array($value)) {
+            if ($this->nullOnEmpty() && (
+                    empty($value['year']) ||
```

Comment:
@Ocramius confirmed - these keys are all defined as part of the date input type in HTML5.